src/libs/actions/Task.js

```diff
@@ -0,0 +1,139 @@
+import Onyx from 'react-native-onyx';
+import CONST from '../../CONST';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as API from '../API';
+import * as ReportUtils from '../ReportUtils';
+import * as Report from './Report';
+import Navigation from '../Navigation/Navigation';
+import ROUTES from '../../ROUTES';
+
+/**
+ * Clears out the task info from the store
+ */
+function clearOutTaskInfo() {
+    Onyx.set(ONYXKEYS.TASK, null);
+}
+
+/**
+ * Assign a task to a user
+ * Function title is createTask for consistency with the rest of the actions
+ * and also because we can create a task without assigning it to anyone
+ * @param {String} parentReportID
+ * @param {String} title
+ * @param {String} description
+ * @param {String} assignee
+ *
+ */
+
+function createTaskAndNavigate(currentUserEmail, parentReportID, title, description, assignee) {
+    // Grab the assigneeChatReportID if there is an assignee
+    const assigneeChatReportID = ReportUtils.getChatByParticipants([assignee]).reportID;
+
+    // Optimistically created Action on the parent
+    const optimisticCreatedAction = ReportUtils.buildOptimisticCreatedReportAction(parentReportID);
+
+    // Create the task report
+    const optimisticTaskReport = ReportUtils.buildOptimisticTaskReport(currentUserEmail, assignee, parentReportID, optimisticCreatedAction.reportActionID, title, description);
+
+    // Create the CreatedReportAction on the task
+    const optimisticTaskCreatedAction = ReportUtils.buildOptimisticCreatedReportAction(optimisticTaskReport.reportID);
+
+    // AddCommentReportAction on the parent chat report
+    const AddCommentText = ` created a task: ${title}`;
+    Report.addComment(parentReportID, AddCommentText);
```

Comment:
The text should just be a url to the task
Have to get the environment, so will look similar to this
```
Environment.getEnvironmentURL()
                .then((environmentURL) => {
                   text = `${environmentURL}/r/${reportID}`);
                });
```

Reply:
On a second though, let's not make this change.

Realizing even if we created this optimistically here - we can't replicate this in the back-end since we don't have access to the environment and aren't passing it. What will probably end up doing is creating an AddComment reportAction that has a taskReportID parameter and then somehow parse that to show the task preview. 

In any case, feel free to leave for now

Reply:
For some reason the second comment _only now_ loaded for me, so I did make some changes. I can revert if need be.